pyomo/common/collections/component_map.py

```diff
@@ -68,18 +96,20 @@ def __str__(self):
 
     def __getitem__(self, obj):
         try:
-            return self._dict[id(obj)][1]
+            return self._dict[_hasher[obj.__class__](obj)][1]
         except KeyError:
-            raise KeyError("Component with id '%s': %s" % (id(obj), str(obj)))
+            _id = _hasher[obj.__class__](obj)
+            raise KeyError("Component with id '%s': %s" % (_id, obj))
```

Comment:
This error message isn't so precise anymore, right, because `_id` might not be an id, if `obj` was hashable to begin with?